Tests/Unit/EmogrifierTest.php

```diff
@@ -1763,4 +1763,245 @@ public function emptyMediaQueriesAreRemoved()
+            '<p class="medium">medium</p>' .
+            '<p class="small">small</p>' .
+            '</body></html>');
+
+        $result = $this->subject->emogrify();
+
+        self::assertSame(
+            1,
+            substr_count($result, '<style type="text/css">' . $css . '</style>')
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function multiLineMediaQueryWithUnixLineEndingsIsAppliedOnlyOnce()
+    {
+        $css = "@media all {\n" .
+            ".medium {font-size:18px;}\n" .
+            ".small {font-size:14px;}\n" .
+            '}';
+        $this->subject->setCss($css);
+        $this->subject->setHtml(
+            $this->html5DocumentType . '<html><body>' .
+            '<p class="medium">medium</p>' .
+            '<p class="small">small</p>' .
+            '</body></html>'
+        );
+
+        $result = $this->subject->emogrify();
+
+        self::assertSame(
+            1,
+            substr_count($result, '<style type="text/css">' . $css . '</style>')
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function multipleMediaQueriesAreAppliedOnlyOnce()
+    {
+        $css = "@media all {\n" .
+            ".medium {font-size:18px;\n" .
+            ".small {font-size:14px;}\n" .
+            '}' .
+            "@media screen {\n" .
+            ".medium {font-size:24px;}\n" .
+            ".small {font-size:18px;}\n" .
+            '}';
+        $this->subject->setCss($css);
+        $this->subject->setHtml(
+            $this->html5DocumentType . '<html><body>' .
+            '<p class="medium">medium</p>' .
+            '<p class="small">small</p>' .
+            '</body></html>'
+        );
+
+        $result = $this->subject->emogrify();
+
+        self::assertSame(
+            1,
+            substr_count($result, '<style type="text/css">' . $css . '</style>')
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function mapBackgroundColor2BgColor()
+    {
+        $css = 'body {background-color: red;}';
+        $this->subject->setHtml($this->html5DocumentType . '<html><body></body></html>');
+        $this->subject->setCss($css);
+        $this->subject->enableCss2HtmlMapping();
+        $html = $this->subject->emogrify();
+
+        self::assertContains(
+            '<body bgcolor="red" style="background-color: red;">',
+            $html
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function mapBackgroundNot2BgColor()
+    {
+        $css = 'body {background: url(bg.png) top;}';
+        $this->subject->setHtml($this->html5DocumentType . '<html><body></body></html>');
+        $this->subject->setCss($css);
+        $this->subject->enableCss2HtmlMapping();
+        $html = $this->subject->emogrify();
+
+        self::assertContains(
+            '<body style="background: url(bg.png) top;">',
+            $html
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function mapBackground2BgColor()
```

Comment:
For all these tests, I propose you use a data provider that provides the CSS style rule and the expected HTML attribute(s): https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers

The return annotation for the data provider then should probably be string[][] (an array of arrays of strings), and the test method then will need to respective param annotations.